### PR TITLE
[WIP] GraphQL: Publish the gift_registry branch

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) contains the work from #8172 and #8170.  Depending on the timing, other PRs might be merged into the `gift_registry` branch.

This PR will not be merged until https://github.com/magento/partners-magento2ee/pull/280 is accepted.

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

whatsnew
Added support for gift registries.

* Use the [giftRegistry query](https://devdocs.magento.com/guides/v2.4/graphql/queries/gift-registry.html) to return the contents of the customer's gift registries.
* Search for gift registries by [type](https://devdocs.magento.com/guides/v2.4/graphql/queries/gift-registry-type-search.html), [email](https://devdocs.magento.com/guides/v2.4/graphql/queries/gift-registry-email-search.html), or [ID](https://devdocs.magento.com/guides/v2.4/graphql/queries/gift-registry-id-search.html).
* [Create](https://devdocs.magento.com/guides/v2.4/graphql/mutations/create-gift-registry.html), [update](https://devdocs.magento.com/guides/v2.4/graphql/mutations/update-gift-registry.html), or [delete](https://devdocs.magento.com/guides/v2.4/graphql/mutations/remove-gift-registry.html) gift registries.
* Add, [update](https://devdocs.magento.com/guides/v2.4/graphql/mutations/update-gift-registry-items.html), or [remove](https://devdocs.magento.com/guides/v2.4/graphql/mutations/remove-gift-registry-items.html) items from a gift registry.
* [Add](https://devdocs.magento.com/guides/v2.4/graphql/mutations/add-gift-registry-registrants.html), [update](https://devdocs.magento.com/guides/v2.4/graphql/mutations/update-gift-registry-registrants.html), or [remove](https://devdocs.magento.com/guides/v2.4/graphql/mutations/remove-gift-registry-registrants.html) registrants from a gift registry.
* [Share](https://devdocs.magento.com/guides/v2.4/graphql/mutations/share-gift-registry.html) a gift registry with invitees.

### CODE REVIEW

File: src/_includes/graphql/gift-registry.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+`created_on` | String! | The date on which the gift registry was created. Only the registry owner can access this attribute
+`dynamic_attributes` | `[GiftRegistryDynamicAttribute]` | An array of attributes that define elements of the gift registry. Each attribute is specified as a code-value pair
+`event_name` | String! | The name the customer assigned to the event
+`id` | ID! | The ID assigned to the gift registry
+`items` | [[GiftRegistryItemInterface](#GiftRegistryItemInterface)] | An array of products added to the gift registry
+`message` | String! | The message text the customer entered to describe the event
+`owner_name` | String! | The customer who created the gift registry
+`privacy_settings` | `GiftRegistryPrivacySettings!` | An enum that states whether the gift registry is PRIVATE or PUBLIC. Only the registry owner can access this attribute
+`registrants` | [[GiftRegistryRegistrant](#GiftRegistryRegistrant)]| Contains details about each registrant for the event
+`shipping_address` | CustomerAddress | Contains the customer's shipping address. Only the registry owner can access this attribute
```
Comment (rogyar) [inline src/_includes/graphql/gift-registry.md:14]: Hi @keharper. Could you clarify, please, is there a technical reason of wrapping only some of the types into `backquotes`? 

Thank you.

Comment (rogyar) [reply src/_includes/graphql/gift-registry.md:14]: I.e. `GiftRegistryPrivacySettings!`vs CustomerAddress

Comment (keharper) [reply src/_includes/graphql/gift-registry.md:14]: They were marked that way in one of the original PRs, and I didn't notice it. I'll correct them.

**Comment by dshevtsov** [comment]
Converted to Draft since all work in progress PRs should be in the Draft state.

**Comment by keharper** [comment]
We can merge this PR when issues #8602  and #8603 are resolved.

**Comment by dshevtsov** [comment]
The PR will be processed internally.

**Pull request closed without merging**